src/django/api/tests/test_wage_indicator_provider.py

```diff
@@ -0,0 +1,238 @@
+            'admin_id': self.contributor.admin_id,
+            'is_verified': self.contributor.is_verified,
+            'contrib_type': self.contributor.contrib_type,
+        }
+
+        formatted_data = self.provider._format_data(
+            self.wage_data,
+            contributor_info
+        )
+
+        self.assertIsInstance(formatted_data, dict)
+        self.assertEqual(formatted_data['field_name'], 'wage_indicator')
+        self.assertEqual(formatted_data['contributor'], contributor_info)
+        self.assertIn('value', formatted_data)
+        self.assertIn('raw_values', formatted_data['value'])
+
+        raw_values = formatted_data['value']['raw_values']
+        self.assertEqual(
+            raw_values['living_wage_link_national'],
+            'https://test.example.com/living'
+        )
+        self.assertIn('living_wage_link_national_text', raw_values)
+
+    def test_fetch_data_complete_flow(self):
+        '''Test complete fetch_data flow returns formatted data.'''
+        data = self.provider.fetch_data(self.facility)
+
+        self.assertIsNotNone(data)
+        self.assertEqual(data['field_name'], 'wage_indicator')
+        self.assertIn('contributor', data)
+        self.assertEqual(data['contributor']['id'], self.contributor.id)
+        self.assertIn('value', data)
+
+    def test_fetch_data_no_wage_data(self):
+        '''Test fetch_data returns None when no wage data exists.'''
+        self.facility.country_code = 'ZZ'
+        data = self.provider.fetch_data(self.facility)
+        self.assertIsNone(data)
+
+    def test_fetch_data_no_contributor_assigned(self):
+        '''Test fetch_data returns None when no contributor assigned.'''
+        # Remove contributor from partner field.
+        self.contributor.partner_fields.remove(self.partner_field)
+
+        data = self.provider.fetch_data(self.facility)
+        self.assertIsNone(data)
+
+    def test_fetch_data_partner_field_inactive(self):
+        '''Test fetch_data works when partner field is inactive.'''
+        # Set the partner field to inactive.
+        self.partner_field.active = False
+        self.partner_field.save()
+
+        # Should still work because provider uses get_all_including_inactive.
+        data = self.provider.fetch_data(self.facility)
+        self.assertIsNotNone(data)
+
+    def test_wage_indicator_link_text_config(self):
+        '''
+        Test WageIndicatorLinkTextConfig customizes display text for links.
+        '''
+        # Update existing or create link text config with custom text.
+        custom_text = 'Custom Test Living Wage Text'
+        WageIndicatorLinkTextConfig.objects \
+            .update_or_create(
+                link_type='living_wage_link_national',
+                defaults={'display_text': custom_text}
+            )
+
+        # Clear cache to ensure new config is used.
+        cache.delete('wage_indicator_link_texts')
+
+        # Fetch data through provider.
```

Comment:
According to best practices the comments should answer the question `Why` something is being done a certain way, this seems like it's just describing `What` is being done.